Sources/Swift/Integrations/SessionReplay/SentryOnDemandReplay.swift

```diff
@@ -158,13 +165,14 @@ class SentryOnDemandReplay: NSObject {
         
         var start = dateProvider.date()
         var actualEnd = start
-        
-        for frame in frames {
-            if frame.time < beginning { continue } else if frame.time > end { break }
-            if frame.time < start { start = frame.time }
-            
-            actualEnd = frame.time
-            framesPath.append(frame.imagePath)
+        workingQueue.dispatchSync {
+            for frame in self.frames {
+                if frame.time < beginning { continue } else if frame.time > end { break }
+                if frame.time < start { start = frame.time }
+                
+                actualEnd = frame.time
+                framesPath.append(frame.imagePath)
```

Comment:
`l`: It's not the path of frames, but paths of frames.
```suggestion
                framePaths.append(frame.imagePath)
```